pkgs/applications/misc/nwg-panel/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib, fetchFromGitHub
+, python3Packages, wrapGAppsHook, gobject-introspection
+, gtk-layer-shell, pango, gdk-pixbuf, atk
+# Extra packages called by various internal nwg-panel modules
+, sway             # swaylock, swaymsg
+, systemd          # systemctl
+, wlr-randr        # wlr-randr
+, nwg-menu
+, withBluetoothControl ? true
+, withBrightnessControl ? true, light
+, withAudioControl ? true, pamixer, pulseaudio
+}:
+
+let
+  inherit (lib) optionals;
+  extraPackages = [ nwg-menu sway systemd wlr-randr ]
+    ++ optionals withBrightnessControl [ light ]
+    ++ optionals withAudioControl [ pamixer pulseaudio ]
+    ;
+in
+python3Packages.buildPythonApplication rec {
+  pname = "nwg-panel";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "nwg-piotr";
+    repo = "nwg-panel";
+    rev = "v${version}";
+    hash = "sha256-x5lGVF6eRhOVXrsBatdsiUiWs/+FxRlCtp79zA206RY=";
+  };
+
+  # No tests
+  doCheck = false;
+
+  # Because of wrapGAppsHook
+  strictDeps = false;
+  dontWrapGApps = true;
+
+  buildInputs = [ atk gdk-pixbuf gtk-layer-shell pango ];
+  nativeBuildInputs = [ wrapGAppsHook gobject-introspection ];
+  propagatedBuildInputs =
+    with python3Packages; [ i3ipc netifaces psutil pygobject3 ]
+    ++ optionals withBluetoothControl [ pybluez ];
+
+  postInstall = ''
+    mkdir -p $out/share/{applications,pixmaps}
+    cp $src/nwg-panel-config.desktop $out/share/applications/
+    cp $src/nwg-shell.svg $src/nwg-panel.svg $out/share/pixmaps/
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix XDG_DATA_DIRS : "$out/share"
+    )
+    makeWrapperArgs+=(
+      "''${gappsWrapperArgs[@]}"
+      --suffix PATH : "${lib.makeBinPath extraPackages}"
```

Comment:
```suggestion
      --prefix PATH : "${lib.makeBinPath [ light nwg-menu pamixer pulseaudio sway systemd wlr-randr ]}"
```

Reply:
I opted to use `suffix` to pick up the packages from the user environment if they had them.

Reply:
I don't understand why we want to do this. It would make the package less reproducible but if you think it is a good idea.